tests/python/nightly/quantization/test_quantization_accuracy.py

```diff
@@ -0,0 +1,151 @@
+        label = gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0)
+        return data, label
+
+    img_size = 299 if model_name == 'inceptionv3' else 224
+    val_data = mx.io.ImageRecordIter(
+        path_imgrec         = rec_val,
+        preprocess_threads  = num_workers,
+        shuffle             = False,
+        batch_size          = batch_size,
+        resize              = 256,
+        data_shape          = (3, img_size, img_size),
+        mean_r              = mean_rgb[0],
+        mean_g              = mean_rgb[1],
+        mean_b              = mean_rgb[2],
+        std_r               = std_rgb[0],
+        std_g               = std_rgb[1],
+        std_b               = std_rgb[2],
+    )
+    return val_data, batch_fn
+
+
+def get_model(model_name, batch_size, qconfig, target=None, original=False, simulated=False):
+    gluon_model = gluon.model_zoo.vision.get_model(model_name, pretrained=True)
+    img_size = 299 if model_name == 'inceptionv3' else 224
+    data_shape = (batch_size, 3, img_size, img_size)
+    mod, params = relay.frontend.from_mxnet(gluon_model, {"data": data_shape})
+    net = mod['main']
+
+    with relay.build_config(opt_level=3):
+        qfunc = relay.quantize.prerequisite_optimize(net, params=params)
+    logging.debug('original')
+    logging.debug(qfunc.astext(show_meta_data=False))
+    if original:
+        return qfunc
+
+    with qconfig:
+        logging.debug('current quantize config')
+        logging.debug(qtz.current_qconfig())
+        qfunc = qtz.quantize(qfunc)
+        logging.debug('after quantize')
+        logging.debug(qfunc.astext(show_meta_data=False))
+    return qfunc
+
+
+def eval_acc(model, dataset, batch_fn, target=tvm.target.cuda(), ctx=tvm.gpu(), log_interval=100):
+    with relay.build_config(opt_level=3):
+        graph, lib, params = relay.build(model, target)
+    # create runtime module
+    m = tvm.contrib.graph_runtime.create(graph, lib, ctx)
+    m.set_input(**params)
+
+    # setup evaluaiton metric
+    dataset.reset()
+    batch_size = dataset.batch_size
+    acc_top1 = mx.metric.Accuracy()
+    acc_top5 = mx.metric.TopKAccuracy(5)
+    acc_top1.reset()
+    acc_top5.reset()
+    # Execute
+    for i, batch in enumerate(dataset):
+        data, label = batch_fn(batch, [mx.cpu(0)])
+        m.run(data=data[0].asnumpy())
+        out_arr = m.get_output(0)
+        acc_top1.update(label, [mx.nd.array(out_arr.asnumpy())])
+        acc_top5.update(label, [mx.nd.array(out_arr.asnumpy())])
+
+        if not (i + 1) % log_interval:
+            _, top1 = acc_top1.get()
+            _, top5 = acc_top5.get()
+            nsamples = (i + 1) * batch_size
+            logging.info('[%d samples] validation: acc-top1=%f acc-top5=%f', nsamples, top1, top5)
+    logging.info('[final] validation: acc-top1=%f acc-top5=%f', top1, top5)
+    return top1
+
+def test_quantize_acc(cfg):
+    rec_val = "/scratch/tqchen/imagenet/val.rec"
```

Comment:
should we hard code it here?

Reply:
This script will not be run by current CI now since it is nightly test and we will open another CI for it. For now it will only be used by developers, I guess adding comment here for requiring user to replace it would be fine.